vllm/v1/sample/metadata.py

```diff
@@ -12,25 +12,24 @@
 @dataclass
 class SamplingMetadata:
 
-    temperature: Optional[torch.Tensor]
+    temperature: torch.Tensor
     all_greedy: bool
     all_random: bool
 
-    top_p: Optional[torch.Tensor]
-    top_k: Optional[torch.Tensor]
+    top_p: torch.Tensor
+    top_k: torch.Tensor
 
     generators: dict[int, torch.Generator]
 
     # None means no logprobs, 0 means sampled token logprobs only
     max_num_logprobs: Optional[int]
 
     no_penalties: bool
-    prompt_token_ids: Optional[torch.Tensor]
     frequency_penalties: torch.Tensor
     presence_penalties: torch.Tensor
     repetition_penalties: torch.Tensor
 
-    output_token_ids: list[list[int]]
+    token_ids: Optional[torch.Tensor]
```

Comment:
We may still need to track the prompt and output ids separately since there are logits processor impls which need to differentiate (even some of the built-in penalties).